src/api/libcellml/analysermodel.h

```diff
@@ -0,0 +1,452 @@
+     *
+     * Return the states in the @c Model processed by this @c Analyser.
+     *
+     * @return The states.
+     */
+    std::vector<AnalyserVariablePtr> states() const;
+
+    /**
+     * @brief Get the state at @p index.
+     *
+     * Return the state at the index @p index for the @c Model processed by this
+     * @c Analyser.
+     *
+     * @param index The index of the state to return.
+     */
+    AnalyserVariablePtr state(size_t index) const;
+
+    /**
+     * @brief Get the number of variables in the @c Model.
+     *
+     * Return the number of variables in the @c Model processed by this
+     * @c Analyser.
+     *
+     * @return The number of variables.
+     */
+    size_t variableCount() const;
+
+    /**
+     * @brief Get the variables in the @c Model.
+     *
+     * Return the variables in the @c Model processed by this @c Analyser.
+     *
+     * @return The variables.
+     */
+    std::vector<AnalyserVariablePtr> variables() const;
+
+    /**
+     * @brief Get the variable at @p index.
+     *
+     * Return the variable at the index @p index for the @c Model processed by
+     * this @c Analyser.
+     *
+     * @param index The index of the variable to return.
+     */
+    AnalyserVariablePtr variable(size_t index) const;
+
+    /**
+     * @brief Get the number of equations in the @c Model.
+     *
+     * Return the number of equations in the @c Model processed by this
+     * @c Analyser.
+     *
+     * @return The number of equations.
+     */
+    size_t equationCount() const;
+
+    /**
+     * @brief Get the equations in the @c Model.
+     *
+     * Return the equations in the @c Model processed by this @c Analyser.
+     *
+     * @return The equations.
+     */
+    std::vector<AnalyserEquationPtr> equations() const;
+
+    /**
+     * @brief Get the equation at @p index.
+     *
+     * Return the equation at the index @p index for the @c Model processed by
+     * this @c Analyser.
+     *
+     * @param index The index of the equation to return.
+     */
+    AnalyserEquationPtr equation(size_t index) const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the equality function.
+     *
+     * Test to determine if @c AnalyserModel needs the equality function, return
+     * @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the equality function, @c false
+     * otherwise.
+     */
+    bool needEqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the inequality
+     * function.
+     *
+     * Test to determine if @c AnalyserModel needs the inequality function,
+     * return @c true if it does and @c false otherwise.
+     *
+     * @return @c true if @c AnalyserModel needs the inequality function,
+     * @c false otherwise.
+     */
+    bool needNeqFunction() const;
+
+    /**
+     * @brief Test to determine if @c AnalyserModel needs the lower than
```

Comment:
should "lower than" be "less than" (otherwise `gt` becomes higher than)? also below.

Reply:
Agreed about "less than", but not about "higher than". We are talking about the `lt` and `gt` (and their derivatives) functions, which stand for "less than" indeed and "greater than". Fixed in commit 681209c.